sensors: convert lis2dh to use information coming from dts

avisconti opened the pull request:
Convert lis2dh accelerometer driver to get the device name as well
as i2c/spi slave information from device tree. Updates the build_all
test accordingly. (issue #11605)

Signed-off-by: Armando Visconti <armando.visconti@st.com>

avisconti commented:
This PR is addressing issue #11605.
I verified building driver for both I2C and SPI, but I tested only I2C on zephyr board.

MaureenHelm commented on dts/bindings/sensor/st,lis2dh.yaml:
```diff
@@ -0,0 +1,24 @@
+
+properties:
+    compatible:
+      constraint: "st,lis2dh"
```
@galak, should this be "st,lis2dh-i2c"?

avisconti replied:
@MaureenHelm 
> should this be "st,lis2dh-i2c"

My opinion is that it does make sense indeed, but if this is a general rule that has to be followed there are other sensors (e.g. lsm6dsl) that need to be changed too.

MaureenHelm reviewed (changes requested):
@avisconti Thanks for your help converting this sensor driver to dts. Can you help with other sensors too? See #9962

avisconti commented on drivers/sensor/lis2dh/Kconfig:
```diff
@@ -8,84 +8,29 @@
-choice
+choice LIS2DH_TRIGGER_MODE
 	prompt "Trigger mode"
-	default LIS2DH_TRIGGER_GLOBAL_THREAD
```
ack

avisconti replied:
I think that there are also the BUS_TYPE changes not related to the dts. All these changes (i.e. BUS_TYPE and TRIGGER_MODE choices) are useful for configuration selection in board though, so I'll keep them but in a separate commit.

avisconti commented:
> Thanks for your help converting this sensor driver to dts. 
> Can you help with other sensors too? See #9962

Sure, I can take the ST ones

avisconti committed:
sensors: convert lis2dh to use information coming from dts

Convert lis2dh accelerometer driver to get the device name as well
as i2c/spi slave information from device tree. Updates the build_all
test accordingly. (issue #11605)

Signed-off-by: Armando Visconti <armando.visconti@st.com>

avisconti committed:
drivers/lis2dh: Add symbol to BUS_TYPE and TRIGGER_MODE config choices

Add symbolic names to bus type (LIS2DH_BUS_TYPE) and trigger mode
(LIS2DH_TRIGGER_MODE) in order to be defined in board defconfig
files.

Signed-off-by: Armando Visconti <armando.visconti@st.com>

avisconti commented:
@galak 
I didn't change the yaml for i2c in this re-push. Let us know your opinion and I'll change accordingly if the case.

avisconti commented:
recheck

MaureenHelm commented:
Looks good, but would like to get feedback from @galak regarding the i2c compatible.

avisconti commented:
@galak @MaureenHelm 
If you both agree I would change the i2c comaptible string into "st,lis2dh-i2c".
It make sense and it is a new creation. We may align the others later on.

avisconti commented:
done!
see if it is ok now.
